packages/pyright-internal/src/analyzer/service.ts

```diff
@@ -640,6 +640,7 @@ export class AnalyzerService {
         configOptions.disableTaggedHints = !!commandLineOptions.disableTaggedHints;
 
         configOptions.initializeTypeCheckingMode(commandLineOptions.typeCheckingMode ?? 'standard');
+        configOptions.applyDiagnosticOverrides(commandLineOptions.diagnosticSeverityOverrides);
```

Comment:
This is the fix here. Apply diagnostic overrides from the command line options first, then apply the json from the pyrightconfig.json files.

Reply:
Hmm, is this the behavior we want? Shouldn't a config file always override all of the diagnostic severity settings (both defaults and specified)? In other words, if there is a config file, the diagnostic severities provided by the language server settings should be ignored, right? Otherwise, developers working on a project will see different behaviors based on their own local settings.

Let's discuss when we meet later today.

Reply:
>  In other words, if there is a config file, the diagnostic severities provided by the language server settings should be ignored, right?

This wasn't the case before. They were combined before AFAIK

Reply:
Yeah in the old code:

- pyrightconfig has an explicit value = this value is used regardless of settings.json
- pyrightconfig doesn't have an explicit value = value in settings.json is used
- settings.json doesn't have an explicit value = default value for typechecking mode is used

Reply:
It's even more complicated than what's listed above because typeCheckingMode can be specified in both places.

Let's discuss what behavior we want. I'm not convinced the old behavior is desirable. The current behavior (what's checked in) is more IMO defensible and consistent with what devs will typically desire.

Is there a precedent set by TypeScript here?

Reply:
I can't find similar settings in typescript. Tsconfig.json doesn't have settings that can be overridden in the VS code settings.json file. 

Maybe command line arguments? Is that the equivalent of settings.json?

Reply:
I don't think command-line arguments are equivalent to settings.json. Command-line arguments should always take precedence whereas settings.json (especially when it's the user settings.json), should not override project-specific settings.

Reply:
Hmm, command line doesn't seem the equivalent. For tsc it trumps everything.

I have a tsconfig.json like so:

```json5
{
    "compilerOptions": {
        "allowUnreachableCode": false,
    }
}
```
And then some code like so:

```typescript
function fn(n: number) {
  if (n > 5) {
    return true;
  } else {
    return false;
  }
  return true;
}
```

Running `tsc`, it outputs an error on the last line. 

If I run `tsc --allowUnreachableCode`, it does not output an error.